### PR TITLE
EdgeWorker support log file upload in chunks

### PR DESCRIPTION
**Pull request by AutomationDev85**
# Description

If Firewall checks for max request sizem, Edge Worker can not push log files without running into an Exception. This PR adds the feature to split log file upload into different chunks if max log file chunk size limit was reached.

# Details about changes

* Add new config value which defined max chunk size. push_log_chunk_size
* If push_log_chunk_size value exceeds then upload split rest of log file into new request.

### CODE REVIEW

File: providers/src/airflow/providers/edge/provider.yaml
Comment (jscheffl) [inline providers/src/airflow/providers/edge/provider.yaml]: Following the NGinx documentation in https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size the default is 1MB.

I'd propose to set the default to 512KB

File: providers/src/airflow/providers/edge/cli/edge_command.py
Comment (jscheffl) [inline providers/src/airflow/providers/edge/cli/edge_command.py]: Fallback is not needed if the config entry is in provider YAML defined.
```suggestion
                    push_log_chunk_size = conf.getint("edge", "push_log_chunk_size")
```

File: providers/src/airflow/providers/edge/provider.yaml
Comment (jscheffl) [inline providers/src/airflow/providers/edge/provider.yaml]: The rationale is usually not a firewall but limits on the application gateway.

Can you point as example to nginx documentation in https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size as nginx is used in most K8s ingress deployments, but also give a hint that if HTTP 413 is seen that this needs to be adjusted.

**Review by jscheffl — COMMENTED** [review]
Some minor comments. Otherwise... looks good in general

File: providers/src/airflow/providers/edge/provider.yaml
Comment (AutomationDev85) [reply providers/src/airflow/providers/edge/provider.yaml]: Done

File: providers/src/airflow/providers/edge/cli/edge_command.py
Comment (AutomationDev85) [reply providers/src/airflow/providers/edge/cli/edge_command.py]: done

File: providers/src/airflow/providers/edge/provider.yaml
Comment (AutomationDev85) [reply providers/src/airflow/providers/edge/provider.yaml]: Updated the description.

**Review by jscheffl — APPROVED** [review]

**Comment by jscheffl** [comment]
Thanks for the improvements! That sounds reasonable to add.

**Comment by potiuk** [comment]
Nice. Good one. There are few small things though for the future @AutomationDev85  (and @jscheffl ) - as written at the top of provider.yaml - we usually increment versions and add release notes by the relase manager - not in the PR, mainly because release manager can take decision on how it should be bumped. (MAJOR/MINOR/PATCHELEVEL) - based on all the changes between last release and new.

This is less of a concern for this pre-release provider, but would be good to keep it - especially removal of previous version is a bit concerning - we usually don't do it, and it makes it a possibility that it will break our automation around releases (but I think in  case of pre-release version it's not that important)

**Comment by jscheffl** [comment]
> Nice. Good one. There are few small things though for the future @AutomationDev85 (and @jscheffl ) - as written at the top of provider.yaml - we usually increment versions and add release notes by the relase manager - not in the PR, mainly because release manager can take decision on how it should be bumped. (MAJOR/MINOR/PATCHELEVEL) - based on all the changes between last release and new.
> 
> This is less of a concern for this pre-release provider, but would be good to keep it - especially removal of previous version is a bit concerning - we usually don't do it, and it makes it a possibility that it will break our automation around releases (but I think in case of pre-release version it's not that important)

Yes, I also thought about it. At the moment it is pre-release... and if you noticed the cadence... we have some contributions as we are moving this new provider to production and by every-day usage we try to harden.

I'd propose to continue like this - once we start a "real release cadence" then we must leave this to release manager. If anything breaks in release automation we will stop of course.

**Comment by potiuk** [comment]
> I'd propose to continue like this - once we start a "real release cadence" then we must leave this to release manager. If anything breaks in release automation we will stop of course.

I am fine. But I guess we'll have to get @eladkal be aware of it and possibly fix any issues quickly :)

**Pull request merged**
